meson.md

```diff
@@ -0,0 +1,45 @@
+# Building with Meson
+
+### Developer build
+
+**Install build tools:** `pip install meson ninja cython pythran`
+
+**Generate ninja make files:** `meson build --prefix=$PWD/build`
+
+**Compile:** `ninja -C build`
+
+**Install:** `meson install -C build`
+
+The install step copies the necessary Python files into the build dir to form a complete package.
```

Comment:
PEP 660 editable installs would probably solve this; see https://github.com/FFY00/meson-python/issues/47 for the relevant tracking issue.

Reply:
@lagru On the other hand, meson & ninja build skimage in under a minute on my machine! There will always be quirks when moving to a new build system but, after working with it for a bit, I am confident that Meson is the right tool for the job.

Reply:
@eli-schwartz I wonder if that issue should be closed and a new one opened with the proposed plan. Right now, almost the whole conversation there is moot.

Reply:
It looks like the shortest path to success is to [have meson install symlinks](https://github.com/FFY00/meson-python/issues/47#issuecomment-1101539767) and then to point to the build directory in the package. This should be very achievable; do you know if anything is blocking this, or whether I should look into implementing it?

Reply:
AFAIK nothing is blocking meson-python from implementing editable installs... simply that no one got around to implementing it.

However @FFY00 would know more.

Reply:
Yeah, building *is* a lot faster with ninja. I'm not vetoing this anyway because in the long-term we have to find an alternative to distutils. :shrug: 

Though it's good to know that there is an intention to support the editable install workflow. From SciPy's meson guide that wasn't clear. Thanks @eli-schwartz, you feedback has been very helpful!

Reply:
@eli-schwartz Until we get editable builds installed, do you know if there is a way to specify in `pyproject.toml` that you want the default build system to be meson, but that you want editable installs to go via setuptools.  Obviously not a long term solution, but could be a band-aid for @lagru's use-case.

Reply:
I don't think pyproject.toml has any support whatsoever for allowing two build systems, there's no thought given to things like "perhaps we want to switch build systems and leave the old one as a fallback during a transition period".

You can invoke the setuptools one with `python setup.py develop`, which editable installs use under the hood for setuptools (this is why a PEP was needed to define what an editable install actually is).

@rgommers can confirm this but iirc SciPy "solved" it by having two pyproject.toml files and copying the backup one in place if you wanted to switch build systems.

Reply:
> You can invoke the setuptools one with `python setup.py develop`,

Yep, that's the right solution for as long as you want to keep `setup.py`'s around

> @rgommers can confirm this but iirc SciPy "solved" it by having two pyproject.toml files and copying the backup one in place if you wanted to switch build systems.

Almost, but not exactly. We recommend packagers who want to do this to simply delete the one `build-backend` line in `pyproject.toml`. That way the build falls back to the default (= setuptools. 

There's a significant flaw in Pip's transition plan for PEP 517 users. I thought we could ship two build systems, and do `pip install --no-use-pep517`, but that doesn't work (for <reasons>). So you can have two build systems in your source tree, but without a patch you can only use one through Pip.

Reply:
Thanks for clarifying. :) It was more or less like I remembered. I suppose editing the pyproject.toml file vs. copying a different one in place isn't all that different.